software/contrib/egressus_melodiam.py

```diff
@@ -0,0 +1,1091 @@
+CV shapes are created using interpolation formulas that fill in the gaps between CV step values in a CV pattern.
+Each interpolation formula is generated by an associated slew shape function.
+There are both linear and non-linear interpolation functions that create various smooth and no-so-smooth shapes.
+
+Each slew shape function generates an array of values (samples) between two given points.
+Samples are output on the CV outputs based on the list of pre-computed interpolated values (sample buffers)
+A value from the sample buffer is retrieved using a Python generator function (self.slewGenerator()).
+One instance of the generator function for each each slew shape exists in a list: self.slewGeneratorObjects[]
+
+A new sample buffer (array of interpolated values) is created at each clock step.
+The number of samples required in each sample buffer (one for each output) is calculated at each clock step or if
+the clock rate or output division changes.
+
+The timing and associated hanlding of input clock interrupts (clocks to din) is not perfect - python/micropython is not
+known to be great at sub-second timing!
+Therefore, at the end of each clock step the algorithm tunes itself to work around sample buffer overruns or underruns.
+If there is a buffer underrun (not enough samples in the buffer), the previous output voltage (sample) is used until the algorithm
+catches back up with itself.
+
+slewGeneratorObjects[] contains 6 object (one for each output)
+Each slewGeneratorObjects[] object is a reference to a copy of the self.slewGenerator() function.
+The self.slewGenerator() function receives a buffer filled with samples and yields one sample each time it is called using next()
+
+In order to maintain the best balance of smooth waves and Rpi pico memory usage, an algorithm is used to vary the 
+sample rate automatically based on the selected output division.
+This causes the sample rate to be at minium when there is a slow clock and high output division.
+Conversely, when there is a fast clock and low output division a higher sample rate is used to avoid unwanted steps
+often caused by under-samping a wave form.
+
+"""
+# Minimum allowed time between incoming 16th note clocks. Smaller values are too fast for poor ole micropython to keep up
+MIN_CLOCK_TIME_MS = 50
+
+# Maximum allowed time between incoming 16th note clocks. This is 2 BPM, larger value causes memory issues unless the next two vals are halfed
+MAX_CLOCK_TIME_MS = 3750
+
+# Max sample rate and clock divisor to keep required buffer sizes within memory limits
+MAX_SAMPLE_RATE = 32
+MAX_OUTPUT_DENOMINATOR = 8
+
+# Calculate maximum sample buffer size required
+SLEW_BUFFER_SIZE_IN_SAMPLES = int(
+    (MAX_CLOCK_TIME_MS / 1000) * MAX_SAMPLE_RATE * MAX_OUTPUT_DENOMINATOR
+)
+
+# How much to print to the console?
+DEBUG_MODE = 0  # 0:Nothing, 1:Some things, 2:Lots, 3: Maybe a bit too much for some
+
+# Attempt to avoid knob hysteresis
+KNOB_CHANGE_TOLERANCE = 0.999
+
+MAX_CV_VOLTAGE = 5

```

Comment:
> MAX_CV_VOLTAGE is also used to determine the maximum voltage that generated CV patterns will contain.

For this, the config includes a variable for "max_output_voltage" which could be used here. If you choose not to then the script will work, but voltages will be clippedt to this value by `europi.py` so it's nice to take it into account per script just in case anyone does set their config max voltage lower than 5V

Reply:
Updated to use the global max_output_voltage config parameter as a reference

Reply:
After some testing and reading of manuals for modules that have CV inputs, I settled on a 5V max. Note that these are not gates or triggers, so cv.voltage() is more appropriate in this case rather than cv.on().